components/localize/localize-mixin.js

```diff
@@ -0,0 +1,166 @@
+import IntlMessageFormat from 'intl-messageformat/src/main.js';
+window.IntlMessageFormat = IntlMessageFormat;
+
+export const D2LLocalizeMixin = superclass => class extends superclass {
+
+	static get properties() {
+		return {
+			language: {type: String},
+			langResources: { type: Object },
+			formats: {
+				type: Object,
+				value: function() {
+					return {
+					}
+				}
+			},
+			useKeyIfMissing: {type: Boolean, value: false},
```

Comment:
Probably a feature we don't need? We could likely throw or log or do something if a translated value for a particular key is missing.